snap/validate.go

```diff
@@ -312,6 +504,13 @@ func ValidateApp(app *AppInfo) error {
 		}
 	}
 
+	if err := validateAppStartupOrdering(app, orderBefore); err != nil {
```

Comment:
Per note above, that may be validateAppOrderNames(app, app.Before) and the one at the top may be validateAppOrderCycles(info.Apps).